### DISCUSSION TITLE
Authentication for sources in azd template source

### DISCUSSION THREAD

**Discussion by lechnerc77** [Azure/azure-dev#2960] [Q&A] [answered]
We can change the list of available template sources via `azd template source add`. Assuming I want to add my custom source I can define a JSON and reference it via e.g.:

```
azd template source add my-source --type url --location https://raw.githubusercontent.com/lechnerc77/azd-festive-tech-calendar-2023/main/custom_template/dapr_template.json
```
How can I use this command and access private GitHub repositories that require authentication?

**Comment by wbreza** [comment, marked as answer]
Today the `url` type template sources don't support authenticated endpoints but would urge you to log an issue as this is a valid next step.  The best we could offer today would be to leverage something like a SAS token based url in an Azure Storage Blob container.

**Reply by lechnerc77** [reply]
Thanks for the answer, so I did not miss anything. I opened the issue #2996 from this discussion
